### ISSUE TITLE
Callback support for HiGHS

### ISSUE THREAD

**Issue by rlloretb** [metab0t/PyOptInterface#17]
Following our previous discussion on HiGHS repo, can callback be supported for HiGHs as well?
I can help with testing.

**Comment by metab0t** [comment]
Yes. I read the documentation of HiGHS callback recently. If I understand correctly, the capability of HiGHS callback is quite limited, we can only query some information or interrupt optimization.

Can you tell me more about your usage of callback of HiGHS and expectation of the functionality?

**Comment by rlloretb** [comment]
My interest is in adding user cuts at the root node to tighten the relaxation. I asking HiGHs devs if it is actually implemented. It is not clear from the documentation if cuts are possible with the current callback entry points.

**Comment by metab0t** [comment]
It seems that user cut and lazy constraint haven't been implemented in HiGHS currently.

I will leave this issue open and get back to this once HiGHS implements these features.

**Comment by rlloretb** [comment]
Great. I asked them and the devs put it as feature request.

https://github.com/ERGO-Code/HiGHS/issues/1837

**Comment by staadecker** [comment]
Fyi, I've gotten a similar request for Gurobi from my team.

**Comment by metab0t** [comment]
> Fyi, I've gotten a similar request for Gurobi from my team.

https://metab0t.github.io/PyOptInterface/callback.html

Callback is supported for Gurobi
